src/Engine/ProtoCore/ClassTable.cs

```diff
@@ -216,9 +216,22 @@ public ProcedureNode GetMemberFunction(string procName, List<ProtoCore.Type> arg
             if (ProcTable == null)
                 return null;
 
-            ProcedureNode procNode = null;
+            ProcedureMatchOptions opts = new ProcedureMatchOptions()
+            {
+                FunctionName = procName,
+                ParameterTypes = argTypeList,
+                ExcludeAutoGeneratedThisProc = true,
+                ExactMatchWithNumArgs = false,
+                ExactMatchWithArgTypes = false
+            };
+
+            if (isStaticOrConstructor)
+            {
+                opts.IsStatic = true;
+                opts.IsConstructor = true;
```

Comment:
Why are both true here?

Reply:
Nice catch, I refactored the code to support one condition or the other